src/sources/splunk_hec.rs

```diff
@@ -789,18 +790,21 @@ mod tests {
 
     /// Splunk token
     const TOKEN: &str = "token";
+    const VALID_TOKENS: &[&str; 2] = &[TOKEN, "secondary-token"];
 
     async fn source() -> (mpsc::Receiver<Event>, SocketAddr) {
-        source_with(Some(TOKEN.to_owned())).await
+        source_with(Some(TOKEN.to_owned()), None).await
     }
 
-    async fn source_with(token: Option<String>) -> (mpsc::Receiver<Event>, SocketAddr) {
+    async fn source_with(token: Option<String>, valid_tokens: Option<&[&str]>) -> (mpsc::Receiver<Event>, SocketAddr) {
```

Comment:
I think we could probably just rewrite this function to _only_ take `valid_tokens` and drop the `token` parameter.

Reply:
If we do that, the tests (using `source_with`) will never cover the case where `SplunkConfig.token` is used. Would that be ok?

Reply:
I think that's ok. We could cover supporting that field with a more focused test that just tests that `token` is inserted into `valid_tokens`.